### PR TITLE
gh-125289: Update sample code in asyncio-task.rst

### PR DESCRIPTION
**Pull request by galmyk**
# gh-125289: Update sample code in asyncio-task.rst

This will change **coroutines** sample code in the **Awaitables** section and make the example clearer.

----
📚 Documentation preview 📚: https://cpython-previews--125292.org.readthedocs.build/

### CODE REVIEW

**Review by JacobCoffee — APPROVED** [review]
this lgtm, but more broadly: i find this example weird in general that if i as a new user copy and paste it it just throws an error.

i think it would be nice to scour the documentation and "split" examples into working/non-working and denote them as such... @willingc thoughts?

**Comment by galmyk** [comment]
@JacobCoffee I agree with splitting working and not working codes. But in this case, code works, just prints a warning:

```
/home/gmt/a.py:10: RuntimeWarning: coroutine 'nested' was never awaited
  nested()  # will raise a "RuntimeWarning".
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
42
```
I'm not native english speaker, So I fear my wording in the following line's comment is not correct:
```python
ested()  # will raise a "RuntimeWarning".
```
It just prints the warning! If it is wrong, please propose a better comment.
Or is it better to remove that command?

**Comment by willingc** [comment]
> this lgtm, but more broadly: i find this example weird in general that if i as a new user copy and paste it it just throws an error.

> i think it would be nice to scour the documentation and "split" examples into working/non-working and denote them as such... @willingc thoughts?

Hmm... there are a couple of concepts at work here: async/await, nesting coroutines, and example behavior

I think the use of `main()` provides a bit more confusion since it is an async function in the example to illustrate nesting. It may be better to use something other than `main` maybe `outer()`.

Where `print()` is placed is a bit of a red herring, since it's the async/await nesting that governs behavior.

I do think it makes sense to improve this section of docs, but I want to make sure that we do it in a way that gets the point across clearly about nesting.

**Comment by galmyk** [comment]
@willingc , @JacobCoffee 
Isn't better to merge this with minimal corrections, and add a more complete change to backlog? I think we need to check other code examples, too.

File: Doc/library/asyncio-task.rst [hunk 1]
```diff
@@ -152,16 +152,16 @@ other coroutines::
     import asyncio
 
     async def nested():
-        return 42
+        print(42)
```
Comment (willingc) [inline Doc/library/asyncio-task.rst]:
```suggestion
        return 42
```

File: Doc/library/asyncio-task.rst [hunk 2]
```diff
@@ -152,16 +152,16 @@ other coroutines::
...
     async def main():
         # Nothing happens if we just call "nested()".
         # A coroutine object is created but not awaited,
         # so it *won't run at all*.
-        nested()
+        nested()  # will raise a "RuntimeWarning".
 
         # Let's do it differently now and await it:
-        print(await nested())  # will print "42".
+        await nested()  # will print "42".
```
Comment (willingc) [inline Doc/library/asyncio-task.rst]:
```suggestion
        print(await nested())  # will print "42".
```

**Review by willingc — COMMENTED** [review]
Let's revert the change in print and limit this PR to adding the runtime error comment.

**Review by willingc — APPROVED** [review]
Thanks @galmyk for the PR and @JacobCoffee for the initial review and discussion.

**Pull request merged**